// @hyperapp/html/dist/html.js
import { h } from "hyperapp"

function vnode(tag) {
  return function (props, children) {
    return typeof props === "object" && !Array.isArray(props)
      ? h(tag, props, children)
      : h(tag, {}, props)
  }
}


export function a(props, children) {
  return vnode("a")(props, children)
}

export function abbr(props, children) {
  return vnode("abbr")(props, children)
}

export function address(props, children) {
  return vnode("address")(props, children)
}

export function area(props, children) {
  return vnode("area")(props, children)
}

export function article(props, children) {
  return vnode("article")(props, children)
}

export function aside(props, children) {
  return vnode("aside")(props, children)
}

export function audio(props, children) {
  return vnode("audio")(props, children)
}

export function b(props, children) {
  return vnode("b")(props, children)
}

export function bdi(props, children) {
  return vnode("bdi")(props, children)
}

export function bdo(props, children) {
  return vnode("bdo")(props, children)
}

export function blockquote(props, children) {
  return vnode("blockquote")(props, children)
}

export function br(props, children) {
  return vnode("br")(props, children)
}

export function button(props, children) {
  return vnode("button")(props, children)
}

export function canvas(props, children) {
  return vnode("canvas")(props, children)
}

export function caption(props, children) {
  return vnode("caption")(props, children)
}

export function cite(props, children) {
  return vnode("cite")(props, children)
}

export function code(props, children) {
  return vnode("code")(props, children)
}

export function col(props, children) {
  return vnode("col")(props, children)
}

export function colgroup(props, children) {
  return vnode("colgroup")(props, children)
}

export function data(props, children) {
  return vnode("data")(props, children)
}

export function datalist(props, children) {
  return vnode("datalist")(props, children)
}

export function dd(props, children) {
  return vnode("dd")(props, children)
}

export function del(props, children) {
  return vnode("del")(props, children)
}

export function details(props, children) {
  return vnode("details")(props, children)
}

export function dfn(props, children) {
  return vnode("dfn")(props, children)
}

export function dialog(props, children) {
  return vnode("dialog")(props, children)
}

export function div(props, children) {
  return vnode("div")(props, children)
}

export function dl(props, children) {
  return vnode("dl")(props, children)
}

export function dt(props, children) {
  return vnode("dt")(props, children)
}

export function em(props, children) {
  return vnode("em")(props, children)
}

export function embed(props, children) {
  return vnode("embed")(props, children)
}

export function fieldset(props, children) {
  return vnode("fieldset")(props, children)
}

export function figcaption(props, children) {
  return vnode("figcaption")(props, children)
}

export function figure(props, children) {
  return vnode("figure")(props, children)
}

export function footer(props, children) {
  return vnode("footer")(props, children)
}

export function form(props, children) {
  return vnode("form")(props, children)
}

export function h1(props, children) {
  return vnode("h1")(props, children)
}

export function h2(props, children) {
  return vnode("h2")(props, children)
}

export function h3(props, children) {
  return vnode("h3")(props, children)
}

export function h4(props, children) {
  return vnode("h4")(props, children)
}

export function h5(props, children) {
  return vnode("h5")(props, children)
}

export function h6(props, children) {
  return vnode("h6")(props, children)
}

export function header(props, children) {
  return vnode("header")(props, children)
}

export function hr(props, children) {
  return vnode("hr")(props, children)
}

export function i(props, children) {
  return vnode("i")(props, children)
}

export function img(props, children) {
  return vnode("img")(props, children)
}

export function input(props, children) {
  return vnode("input")(props, children)
}

export function ins(props, children) {
  return vnode("ins")(props, children)
}

export function kbd(props, children) {
  return vnode("kbd")(props, children)
}

export function label(props, children) {
  return vnode("label")(props, children)
}

export function legend(props, children) {
  return vnode("legend")(props, children)
}

export function li(props, children) {
  return vnode("li")(props, children)
}

export function main(props, children) {
  return vnode("main")(props, children)
}

export function map(props, children) {
  return vnode("map")(props, children)
}

export function mark(props, children) {
  return vnode("mark")(props, children)
}

export function menu(props, children) {
  return vnode("menu")(props, children)
}

export function menuitem(props, children) {
  return vnode("menuitem")(props, children)
}

export function meter(props, children) {
  return vnode("meter")(props, children)
}

export function nav(props, children) {
  return vnode("nav")(props, children)
}

export function object(props, children) {
  return vnode("object")(props, children)
}

export function ol(props, children) {
  return vnode("ol")(props, children)
}

export function optgroup(props, children) {
  return vnode("optgroup")(props, children)
}

export function option(props, children) {
  return vnode("option")(props, children)
}

export function output(props, children) {
  return vnode("output")(props, children)
}

export function p(props, children) {
  return vnode("p")(props, children)
}

export function param(props, children) {
  return vnode("param")(props, children)
}

export function pre(props, children) {
  return vnode("pre")(props, children)
}

export function progress(props, children) {
  return vnode("progress")(props, children)
}

export function q(props, children) {
  return vnode("q")(props, children)
}

export function rp(props, children) {
  return vnode("rp")(props, children)
}

export function rt(props, children) {
  return vnode("rt")(props, children)
}

export function rtc(props, children) {
  return vnode("rtc")(props, children)
}

export function ruby(props, children) {
  return vnode("ruby")(props, children)
}

export function s(props, children) {
  return vnode("s")(props, children)
}

export function samp(props, children) {
  return vnode("samp")(props, children)
}

export function section(props, children) {
  return vnode("section")(props, children)
}

export function select(props, children) {
  return vnode("select")(props, children)
}

export function small(props, children) {
  return vnode("small")(props, children)
}

export function source(props, children) {
  return vnode("source")(props, children)
}

export function span(props, children) {
  return vnode("span")(props, children)
}

export function strong(props, children) {
  return vnode("strong")(props, children)
}

export function sub(props, children) {
  return vnode("sub")(props, children)
}

export function summary(props, children) {
  return vnode("summary")(props, children)
}

export function sup(props, children) {
  return vnode("sup")(props, children)
}

export function svg(props, children) {
  return vnode("svg")(props, children)
}

export function table(props, children) {
  return vnode("table")(props, children)
}

export function tbody(props, children) {
  return vnode("tbody")(props, children)
}

export function td(props, children) {
  return vnode("td")(props, children)
}

export function textarea(props, children) {
  return vnode("textarea")(props, children)
}

export function tfoot(props, children) {
  return vnode("tfoot")(props, children)
}

export function th(props, children) {
  return vnode("th")(props, children)
}

export function thead(props, children) {
  return vnode("thead")(props, children)
}

export function time(props, children) {
  return vnode("time")(props, children)
}

export function tr(props, children) {
  return vnode("tr")(props, children)
}

export function track(props, children) {
  return vnode("track")(props, children)
}

export function u(props, children) {
  return vnode("u")(props, children)
}

export function ul(props, children) {
  return vnode("ul")(props, children)
}

export function video(props, children) {
  return vnode("video")(props, children)
}

export function vvar(props, children) {
  return vnode("vvar")(props, children)
}

export function wbr(props, children) {
  return vnode("wbr")(props, children)
}
